### DISCUSSION TITLE
warning WebpackError: Unsafe builtin usage http.Agent in Gatsby 3.0

### DISCUSSION THREAD

**Discussion by ppatel31** [gatsbyjs/gatsby#30443] [Help]
## Description

warning WebpackError: Unsafe builtin usage http.Agent
--
13 | at my-website/node_modules/@aws-sdk/node-http-handler/dist/es/node-http-handler.js:19:1
14 | at my-website/node_modules/@aws-sdk/client-lex-runtime-service/dist/es/runtimeConfig.js:20:153
15 | at my-website/webpack/bootstrap:19:1
16 | at my-website/webpack/bootstrap:19:1
17 | at my-website/webpack/bootstrap:19:1
18 | at my-website/webpack/bootstrap:19:1
19 | at my-website/webpack/bootstrap:19:1
20 | at my-website/webpack/bootstrap:19:1
21 | at my-website/.cache/_this_is_virtual_fs_path_/$virtual/sync-requires.js:175:67
22 | at my-website/webpack/bootstrap:19:1
23 | at my-website/.cache/static-entry.js:11:22
24 |  
25 | 17 \|         var keepAlive = true;
26 | 18 \|         var maxSockets = 50;
27 | > 19 \|         this.httpAgent = httpAgent \|\| new hAgent({ keepAlive: keepAlive, maxSockets: maxSockets });
28 | \| ^
29 | 20 \|         this.httpsAgent = httpsAgent \|\| new hsAgent({ keepAlive: keepAlive, maxSockets: maxSockets });
30 | 21 \|     }
31 | 22 \|     NodeHttpHandler.prototype.destroy = function () {

### Steps to reproduce

Updated 2.0 to 3.0

Run Gatsby Build

How to Make a Minimal Reproduction: https://www.gatsbyjs.com/contributing/how-to-make-a-reproducible-test-case/

Works but get 5 of above warnings

What should happen?

Works but need to resolve this warning

What happened.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.13.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.90
    Edge: 89.0.774.57
    Firefox: 86.0
    Safari: 14.0.3
  npmPackages:
    gatsby: ^3.1.0 => 2.28.0
    gatsby-plugin-canonical-urls: ^3.1.0 => 2.6.0
    gatsby-plugin-google-tagmanager: ^3.1.0 => 2.7.0
    gatsby-plugin-preconnect: ^1.2.0 => 1.1.54
    gatsby-plugin-react-helmet: ^4.1.0 => 3.6.0
    gatsby-plugin-remove-trailing-slashes: ^3.1.0 => 2.6.0
    gatsby-plugin-robots-txt: ^1.5.5 => 1.5.3
    gatsby-plugin-sitemap: ^3.1.0 => 2.8.0
    gatsby-plugin-verify-bing: ^1.1.26 => 1.1.24
    gatsby-remark-external-links: 0.0.4 => 0.0.4
    gatsby-source-filesystem: ^3.1.0 => 2.7.0
    gatsby-source-graphql: ^3.1.0 => 2.10.0
    gatsby-transformer-json: ^3.1.0 => 2.7.0
    gatsby-transformer-remark: ^3.1.0 => 2.12.0
  npmGlobalPackages:
    gatsby-cli: 3.1.0
    gatsby: 2.19.41

**Comment by LekoArts** [comment]
This was fixed in https://github.com/gatsbyjs/gatsby/pull/30216 and `gatsby@3.1.2`

**Reply by iDVB** [reply]
Having this same issue in Gatsby 3 when trying to use [git-repo-info](https://www.npmjs.com/package/git-repo-info) in **gatsby-ssr.js** in order to inject the git sha into our html. 

Is there something external packages need to do to adhere or is this just Gatsby being overly cautious?

**Reply by alana314** [reply]
We have a similar issue using apollo. We have a custom preview instance that uses apollo to preview our content. Gatsby 3.9.1 detects this as an unsafe builtin and disables incremental builds because of UNSAFE_BUILTIN. Fix #30216 doesn't fix this:
```
warning Unsafe builtin method was used, future builds will need to rebuild all pages
warning WebpackError: Unsafe builtin usage https.request: 
    at site/node_modules/isomorphic-fetch/node_modules/node-fetch/index.js:119:1
    at site/node_modules/isomorphic-fetch/node_modules/node-fetch/index.js:49:1
    at site/node_modules/isomorphic-fetch/node_modules/node-fetch/index.js:37:1
    at site/node_modules/isomorphic-fetch/fetch-npm-node.js:8:1
    at site/node_modules/apollo-link-http/lib/bundle.esm.js:69:1
    at site/node_modules/zen-observable/lib/Observable.js:197:1
    at site/node_modules/zen-observable/lib/Observable.js:279:1
    at site/node_modules/apollo-link-error/lib/bundle.esm.js:11:1
    at site/node_modules/zen-observable/lib/Observable.js:197:1
    at site/node_modules/zen-observable/lib/Observable.js:279:1
    at site/node_modules/apollo-client/bundle.esm.js:864:1
    at site/node_modules/zen-observable/lib/Observable.js:197:1
    at site/node_modules/zen-observable/lib/Observable.js:279:1
    at site/node_modules/apollo-client/bundle.esm.js:1595:1
    at site/node_modules/apollo-client/bundle.esm.js:1630:1
```

**Reply by hbeckmann** [reply]
Experiencing the same error -- we are using @apollo/client with gatsby-ssr and gatsby-browser, I'm actually trying to debug an issue where our pages aren't building with any static HTML in the body. I noticed other people had similar issues with providers on their gatsby-ssr file, I cannot find anything related to apollo though.

It appears you are using isomorphic-fetch while we are using cross-fetch though @alana314
